scripts/translate_transcript.py

```diff
@@ -2,65 +2,78 @@
 import argparse
 import asyncio
 import json
+from pathlib import Path
 from unittest.mock import AsyncMock
 
 from mentat.code_file_manager import CODE_FILE_MANAGER, CodeFileManager
 from mentat.git_handler import GIT_ROOT
 from mentat.parsers.block_parser import BlockParser
+from mentat.parsers.git_parser import GitParser
 from mentat.parsers.parser import Parser
 from mentat.parsers.replacement_parser import ReplacementParser
 from mentat.parsers.split_diff_parser import SplitDiffParser
 from mentat.parsers.unified_diff_parser import UnifiedDiffParser
 from mentat.session_stream import SESSION_STREAM
 from mentat.utils import convert_string_to_asyncgen
 
-CODE_FILE_MANAGER.set(CodeFileManager())
-SESSION_STREAM.set(AsyncMock())
-
-
 parser_map: dict[str, Parser] = {
     "block": BlockParser(),
     "replacement": ReplacementParser(),
     "split-diff": SplitDiffParser(),
     "unified-diff": UnifiedDiffParser(),
+    "git": GitParser(),
 }
 
-parser = argparse.ArgumentParser(
-    description="Translate transcript between parsing formats"
-)
-parser.add_argument(
-    "--transcript", type=str, default=None, help="Transcript to translate"
-)
-# TODO: infer from config or something
-parser.add_argument(
-    "--starting-format",
-    type=str,
-    default="block",
-    help="Format of the transcript to translate",
-)
-parser.add_argument(
-    "--ending-format", type=str, default="block", help="Format to translate to"
-)
-parser.add_argument("--git-root", type=str, default=".", help="Git root directory")
-args = parser.parse_args()
-GIT_ROOT.set(args.git_root)
 
-starting_parser = parser_map[args.starting_format]
-ending_parser = parser_map[args.ending_format]
+def translate_message(message: str, starting_parser, ending_parser) -> str:
+    if hasattr(starting_parser, "parse_string"):
+        parsedLLMResponse = starting_parser.parse_string(message)
```

Comment:
We could get rid of this hasattr if we made a stream_and_parse_llm_response that just collected the async generator back into a string and passed it to parse_string; just an idea though, if you don't like it no need to do it (it is a bit weird converting a string to a generator and back haha

Reply:
That's probably a good idea. I'll do that. Could also see giving parsers a parse_string method since it might be useful in other contexts and makes more sense

Reply:
Maybe, but I think that would take a lot of work making a parse_string method for all of the parsers, since it's so interlinked with streaming right now.

Reply:
Yeah, you're right. I did it your way.